tests/components/heos/test_config_flow.py

```diff
@@ -244,6 +246,143 @@ async def test_discovery_updates(
     assert config_entry.data[CONF_HOST] == "127.0.0.2"
 
 
+async def test_zeroconf_discovery(
```

Comment:
Copy/pasted from original `test*discovery*` methods. Only changes are to `discovery_data*` parameters which are also `zeroconf_` prefixed and obviously updated the type. And `async_init` is now invoked with `source: SOURCE_ZEROCONF` (instead of SSDP) for obvious reasons 😃